diff --git a/dist/vue-repl.css b/dist/vue-repl.css
index d9f07c16f570d381b3d2e50c054ae256831dd69a..3e22348ba61937000aaca360786b0ff63ad51266 100644
--- a/dist/vue-repl.css
+++ b/dist/vue-repl.css
@@ -35,7 +35,7 @@
   color: var(--text-light);
   position: absolute;
   left: 50%;
-  bottom: 20px;
+  bottom: 17px;
   background-color: var(--bg);
   padding: 8px 12px;
   border-radius: 8px;
@@ -231,6 +231,7 @@ button.active[data-v-f221f6e0] {
   margin-right: 10px;
 }
 .file[data-v-f661725b] {
+  position: relative;
   display: inline-block;
   font-size: 13px;
   font-family: var(--font-code);
@@ -248,18 +249,18 @@ button.active[data-v-f221f6e0] {
   padding: 8px 10px 6px;
   line-height: 20px;
 }
+.file.pending span[data-v-f661725b] {
+  min-width: 50px;
+  padding-right: 32px
+}
 .file.pending input[data-v-f661725b] {
-  width: 90px;
-  height: 30px;
-  line-height: 30px;
+  position: absolute;
+  inset: 8px 6px 9px;
+  line-height: 20px;
   outline: none;
-  border: 1px solid var(--border);
-  border-radius: 4px;
-  padding: 0 0 0 10px;
-  margin-top: 2px;
-  margin-left: 6px;
-  font-family: var(--font-code);
-  font-size: 12px;
+  border: none;
+  padding: 0 2px;
+  min-width: 1px;
 }
 .file .remove[data-v-f661725b] {
   display: inline-block;
@@ -306,8 +307,8 @@ button.active[data-v-f221f6e0] {
 
 .wrapper[data-v-70b24951] {
   position: absolute;
-  bottom: 8px;
-  right: 15px;
+  bottom: 21px;
+  left: 15px;
   z-index: 11;
   display: flex;
   align-items: center;
diff --git a/dist/vue-repl.js b/dist/vue-repl.js
index 76ec54386cbda4ea0e0c054f1373d0a056a54d09..ebc371d3180a202473aaf3f227ea119b9929cd8d 100644
--- a/dist/vue-repl.js
+++ b/dist/vue-repl.js
@@ -623,6 +623,7 @@ Tip: edit the "Import Map" tab to specify import paths for dependencies.`;
             ...ssrModules,
             `import { renderToString as _renderToString } from 'vue/server-renderer'
          import { createSSRApp as _createApp } from 'vue'
+         ${previewOptions?.customCode?.importCode || ""}
          const AppComponent = __modules__["${mainFile}"].default
          AppComponent.name = 'Repl'
          const app = _createApp(AppComponent)
@@ -630,6 +631,7 @@ Tip: edit the "Import Map" tab to specify import paths for dependencies.`;
            app.config.unwrapInjectedRef = true
          }
          app.config.warnHandler = () => {}
+         ${previewOptions?.customCode?.useCode || ""}
          window.__ssr_promise__ = _renderToString(app).then(html => {
            document.body.innerHTML = '<div id="app">' + html + '</div>' + \`${previewOptions?.bodyHTML || ""}\`
          }).catch(err => {
@@ -17190,6 +17192,12 @@ class File {
     if (this.filename.endsWith(".ts")) {
       return "typescript";
     }
+    if (this.filename.endsWith(".md")) {
+      return "markdown";
+    }
+    if (this.filename.endsWith(".sass") || this.filename.endsWith(".scss")) {
+      return "scss";
+    }
     return "javascript";
   }
 }
@@ -17585,9 +17593,9 @@ const _sfc_main$3 = /* @__PURE__ */ defineComponent({
         return;
       const filename = "src/" + pendingFilename.value;
       const oldFilename = pending.value === true ? "" : pending.value;
-      if (!/\.(vue|js|ts|css|json)$/.test(filename)) {
+      if (!/\.(vue|js|ts|css|json|md|s[ac]ss)$/.test(filename)) {
         store.state.errors = [
-          `Playground only supports *.vue, *.js, *.ts, *.css, *.json files.`
+          `Playground only supports *.vue, *.js, *.ts, *.css, *.json, *.md, *.sass, *.scss files.`
         ];
         return;
       }
@@ -17643,6 +17651,7 @@ const _sfc_main$3 = /* @__PURE__ */ defineComponent({
               }, _hoisted_5, 8, _hoisted_3$1)) : createCommentVNode("", true)
             ], 42, _hoisted_1$3)) : createCommentVNode("", true),
             pending.value === true && i === files.value.length - 1 || pending.value === file ? (openBlock(), createElementBlock("div", _hoisted_6, [
+              createElementVNode("span", _hoisted_2$1, pendingFilename.value, 1),
               withDirectives(createElementVNode("input", {
                 "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => pendingFilename.value = $event),
                 spellcheck: "false",
